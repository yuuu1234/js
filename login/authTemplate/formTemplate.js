import React from 'react';
import {
  SafeAreaView, KeyboardAvoidingView, Platform, ScrollView,
} from 'react-native';
import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import styles from './style';

const withAuthTemplate = ({ Header, Form, Footer }) => {
  @observer
  class Screen extends React.Component {
    handleNavigation = (screenType) => {
      const { navigation } = this.props;
      navigation.navigate(screenType);
    }

    render() {
      return (
        <SafeAreaView style={styles.container}>
          <KeyboardAvoidingView
            style={styles.container}
            behavior={Platform.OS === 'ios' ? 'padding' : null}
          >
            <ScrollView
              showsVerticalScrollIndicator={false}
              keyboardShouldPersistTaps="handled"
              contentContainerStyle={{ flexGrow: 1 }}
            >
              {Header && <Header />}
              {Form && <Form handleNavigation={this.handleNavigation} />}
            </ScrollView>
            {Footer && <Footer handleNavigation={this.handleNavigation} />}
          </KeyboardAvoidingView>
        </SafeAreaView>
      );
    }
  }

  Screen.propTypes = {
    navigation: PropTypes.shape({ navigate: PropTypes.func }).isRequired,
  };

  return Screen;
};

export { withAuthTemplate };
