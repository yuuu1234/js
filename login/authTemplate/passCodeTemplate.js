import React from 'react';
import { View, ScrollView } from 'react-native';
import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import styles from './style';

const withPassCodeTemplate = ({ Header, Main, Footer }) => {
  @observer
  class Screen extends React.Component {
    handleNavigation = (screenType) => {
      const { navigation } = this.props;
      navigation.navigate(screenType);
    }

    render() {
      return (
        <View style={styles.passCodeContainer}>
          <ScrollView showsVerticalScrollIndicator={false} contentContainerStyle={{ flexGrow: 1 }}>
            { Header && <Header /> }
            { Main && <Main handleNavigation={this.handleNavigation} /> }
          </ScrollView>
          { Footer && (
            <View style={styles.passCodeFooter}>
              <Footer handleNavigation={this.handleNavigation} />
            </View>
          ) }
        </View>
      );
    }
  }

  Screen.propTypes = {
    navigation: PropTypes.shape({
      navigate: PropTypes.func.isRequired,
    }).isRequired,
  };

  return Screen;
};

export { withPassCodeTemplate };
