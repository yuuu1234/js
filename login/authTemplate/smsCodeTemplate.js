import React from 'react';
import { KeyboardAvoidingView } from 'react-native';
import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import styles from './style';

const withSMSCodeTemplate = ({
  Header, Main, Footer,
}) => {
  @observer
  class Screen extends React.Component {
    handleNavigation = (screenType) => {
      const { navigation } = this.props;
      navigation.navigate(screenType);
    }

    render() {
      return (
        <KeyboardAvoidingView behavior="padding" enabled style={styles.container}>
          { Header && <Header /> }
          { Main && <Main handleNavigation={this.handleNavigation} /> }
          { Footer && <Footer handleNavigation={this.handleNavigation} /> }
        </KeyboardAvoidingView>
      );
    }
  }

  Screen.propTypes = {
    navigation: PropTypes.shape({ navigate: PropTypes.func }).isRequired,
  };

  return Screen;
};

export { withSMSCodeTemplate };
