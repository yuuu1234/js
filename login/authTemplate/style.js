import { StyleSheet } from 'react-native';
import { Colors } from '@src/styles';

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.White,
    flex: 1,
  },
  passCodeContainer: {
    backgroundColor: Colors.White,
    flex: 1,
  },
  passCodeFooter: {
    marginBottom: 20,
  },
  resendCodeContainer: {
    paddingBottom: 50,
    textAlign: 'center',
  },
  resendContainer: {
    padding: 5,
  },
  resendButton: {
    textDecorationLine: 'underline',
    fontSize: 16,
    color: Colors.GreyText,
    textAlign: 'center',
  },
});

export default styles;
