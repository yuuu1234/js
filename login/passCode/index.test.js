import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import GlobalStoreV2 from '@src/stores/globalV2.store';
import PassCodeForm from './passCodeFormFirebase';
import Footer from './footer';
import { withPassCodeTemplate } from '../authTemplate/passCodeTemplate';
import LoginStore from '../store/login.store';

const store = new LoginStore();
const Screen = withPassCodeTemplate({ main: PassCodeForm, Footer });
it('Pass code screen should render correctly', async () => {
  const tree = renderer.create(
    <Provider loginStore={store} globalStoreV2={new GlobalStoreV2()}><Screen /></Provider>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
