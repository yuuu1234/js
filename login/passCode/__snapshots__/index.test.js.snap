// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pass code screen should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    showsVerticalScrollIndicator={false}
  >
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <View>
      Object {
        "children": Array [
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Login
          </Text>,
        ],
        "props": Object {
          "accessibilityLabel": "FormButton",
          "accessible": true,
          "focusable": true,
          "onClick": [Function],
          "onResponderGrant": [Function],
          "onResponderMove": [Function],
          "onResponderRelease": [Function],
          "onResponderTerminate": [Function],
          "onResponderTerminationRequest": [Function],
          "onStartShouldSetResponder": [Function],
          "style": Array [
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#003D9A",
              "borderRadius": 15,
              "height": 41,
              "justifyContent": "center",
              "marginBottom": 16,
              "marginHorizontal": 49,
              "marginTop": 6,
            },
            Object {
              "backgroundColor": "#97B5E2",
            },
            Object {},
          ],
        },
        "type": "View",
      }
      <Text
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "color": "#929292",
            "fontSize": 16,
            "justifyContent": "center",
            "textDecorationLine": "underline",
          }
        }
      >
        Forgot your passcode?
      </Text>
      <Modal
        hardwareAccelerated={false}
        onDismiss={[Function]}
        transparent={true}
        visible={false}
      />
    </View>
  </View>
</View>
`;
